tests/unit/botocore/test_config_provider.py

```diff
@@ -556,3 +557,206 @@ def test_provide_section_config_with_only_overrides(self):
                 'override_var': 'override',
             }
         )
+
```

Comment:
We should make sure to port over some of the shallow copy tests for the `ConfigValueStore` in the original PR: https://github.com/boto/botocore/pull/2973/files#diff-0d043ca2ec3f69519455add1a50265e2a7fa3c0296392ea9b8a4c9578794f6d7R377